From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 31 Oct 2021 21:23:35 +0100
Subject: [PATCH] Fix symbol conflict with OpenLDAP 2.5+

    In file included from ldap_connect.c:34:
    ldap_connect.h:34:12: error: conflicting types for 'ldap_connect'
       34 | extern int ldap_connect(char *_ld_name);
          |            ^~~~~~~~~~~~
    In file included from ldap_connect.c:32:
    /usr/include/ldap.h:1555:1: note: previous declaration of 'ldap_connect' was here
     1555 | ldap_connect( LDAP *ld );
          | ^~~~~~~~~~~~

--- a/src/modules/ldap/ldap_connect.c
+++ b/src/modules/ldap/ldap_connect.c
@@ -185,7 +185,7 @@
 	return 0;
 }
 
-int ldap_connect(char *_ld_name)
+int kam_ldap_connect(char *_ld_name)
 {
 	return ldap_connect_ex(_ld_name, L_DBG);
 }
--- a/src/modules/ldap/ldap_connect.h
+++ b/src/modules/ldap/ldap_connect.h
@@ -31,7 +31,7 @@
 #include "../../core/str.h"
 #include "../../core/dprint.h"
 
-extern int ldap_connect(char *_ld_name);
+extern int kam_ldap_connect(char *_ld_name);
 extern int ldap_disconnect(char *_ld_name);
 extern int ldap_reconnect(char *_ld_name);
 extern int ldap_get_vendor_version(char **_version);
--- a/src/modules/ldap/ldap_mod.c
+++ b/src/modules/ldap/ldap_mod.c
@@ -167,7 +167,7 @@
 			return -1;
 		}
 
-		if(ldap_connect(ld_name) != 0) {
+		if(kam_ldap_connect(ld_name) != 0) {
 			LM_ERR("[%s]: failed to connect to LDAP host(s)\n", ld_name);
 			ldap_disconnect(ld_name);
 			return -1;
